### PR TITLE
Show icon on the file to indicate if a file is encrypted or not

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@jancborchardt

### CODE REVIEW

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -745,6 +745,11 @@
 				"href": linkUrl
 			});
 
+			// show encryption icon
+			if (fileData.encrypted) {
```
Comment (DeepDiver1975) [inline apps/files/js/filelist.js]: @jancborchardt remove the if statement to have the icon added in any case

**Comment by karlitschek** [comment]
thats nice :+1:

**Comment by jancborchardt** [comment]
@DeepDiver1975 styling done!

**Comment by DeepDiver1975** [comment]
bildschirmfoto von 2015-04-14 15 31 59

**Comment by jancborchardt** [comment]
Yeah, the overlay isn’t really that good-looking after all with opacity. Let me make a separate icon-encrypted app which is half-grey but fully opaque.

**Comment by jancborchardt** [comment]
@DeepDiver1975 fixed the icon.

**Comment by jancborchardt** [comment]
But to be honest I’m not super-happy about all the files being overlayed with such an indicator. I’m still not sure about the use of it. Maybe what would be better is a sidebar entry for »Non-encrypted files«?

It just feels that this really introduces a very technical level into encryption. But one of our tasks should be to make file encryption easily usable. This however just looks odd. :\

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
> Maybe what would be better is a sidebar entry for »Non-encrypted files«?

Not really - because we cannot really search on this criteria

**Comment by jancborchardt** [comment]
Ok – let’s discuss tomorrow then please what we need this for and if there’s a better solution to the requirement.

**Comment by PVince81** [comment]
I also disagree with having a lock icon on EVERY file, this becomes boring if all of them have it.

If the purpose is only to be able to identify unencrypted files (aka insecure?) in case of a failed migration, then the icon should rather be provided for those, maybe a warning icon.

**Comment by DeepDiver1975** [comment]
> If the purpose is only to be able to identify unencrypted files (aka insecure?) in case of a failed migration, then the icon should rather be provided for those, maybe a warning icon.

well - thinking about the big picture where folder based encryption is possible - this is no longer a that academic issue.

**Comment by nickvergessen** [comment]
Well since it's per folder and not per file, maybe we can add it on folder icons only, and then also add it in the header bar next to the "New" + "Upload" features?

**Comment by jancborchardt** [comment]
That will result in less icons, but it’s still strange.

A general design principle is to only show something special like that if there’s an error, not if it’s the expected case. When encryption is enabled, everything being encrypted is expected. The focus should be on the files which are _not_ encrypted – and how to encrypt them too.

**Comment by DeepDiver1975** [comment]
so what shall we do with this? From an dev/qa perspective there is also the need to know if a file is encrypted of not. Shall we have a setting for this? :me_hides:

**Comment by PVince81** [comment]
You could at first simply set a class on the "tr" and/or add an attribute "data-encrypted=true". If QA want to check if the file is encrypted, a DOM inspector could help.

**Comment by PVince81** [comment]
On the other hand it would also imply that we'd want to return the same data with WebDAV in a new property "oc:encrypted" (considering that in the future the Web UI will use WebDAV too)

**Comment by jancborchardt** [comment]
Yeah, for dev/qa we shouldn’t introduce interface elements. A data attribute would suffice.

I’m still uncomfortable introducing this icon in any case. It requires more discussion on what exactly the use-case is, as well as making sure it doesn’t degrade the UX in such a major way for people who choose to use encryption. Pushing to 8.2 hence.

**Comment by DeepDiver1975** [comment]
> Pushing to 8.2 hence.

agreed

**Pull request closed without merging**